### ISSUE TITLE
Doesn't load at all with Obsidian 1.7.2

### ISSUE THREAD

**Issue by Cammagno** [FlorianWoelki/obsidian-iconize#566]
More details here:
https://discord.com/channels/686053708261228577/716028884885307432/1286388714158948414

**Comment by I-Pch** [comment]
As I also encounter this in one of my vault I'm going to add a bit more info to this, in hope that it could be useful 😊 : 

There's at least an error in the console appearing as soon as Obsidian tries to load `Iconize`: 

**Error 1:**
```
app.js:1 Obsidian Developer Console
plugin:obsidian-icon-folder:6375 loading iconize
app.js:1 TypeError: Cannot read properties of undefined (reading 'get')
    at retrieveLookupItem (plugin:obsidian-icon-folder:5411:41)
    at eval (plugin:obsidian-icon-folder:5430:13)
    at Array.forEach (<anonymous>)
    at BookmarkInternalPlugin.computeNodesWithPath (plugin:obsidian-icon-folder:5429:15)
    at BookmarkInternalPlugin.onMount (plugin:obsidian-icon-folder:5435:14)
    at eval (plugin:obsidian-icon-folder:6603:32)
    at Array.forEach (<anonymous>)
    at IconizePlugin.handleChangeLayout (plugin:obsidian-icon-folder:6601:38)
    at eval (plugin:obsidian-icon-folder:6396:57)
    at t.<anonymous> (app.js:1:2272343)
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
a @ app.js:1
Promise.then (async)
l @ app.js:1
a @ app.js:1
Promise.then (async)
l @ app.js:1
a @ app.js:1
Promise.then (async)
l @ app.js:1
(anonymous) @ app.js:1
v @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
a @ app.js:1
Promise.then (async)
l @ app.js:1
a @ app.js:1
Promise.then (async)
l @ app.js:1
(anonymous) @ app.js:1
v @ app.js:1
t.loadLayout @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
a @ app.js:1
Promise.then (async)
l @ app.js:1
a @ app.js:1
Promise.then (async)
l @ app.js:1
a @ app.js:1
```
**Error 2**
```
app.js:1 Uncaught TypeError: Cannot read properties of undefined (reading 'get')
    at retrieveLookupItem (plugin:obsidian-icon-folder:5411:41)
    at eval (plugin:obsidian-icon-folder:5430:13)
    at Array.forEach (<anonymous>)
    at BookmarkInternalPlugin.computeNodesWithPath (plugin:obsidian-icon-folder:5429:15)
    at BookmarkInternalPlugin.onMount (plugin:obsidian-icon-folder:5435:14)
    at eval (plugin:obsidian-icon-folder:6603:32)
    at Array.forEach (<anonymous>)
    at IconizePlugin.handleChangeLayout (plugin:obsidian-icon-folder:6601:38)
    at eval (plugin:obsidian-icon-folder:6458:82)
    at e.tryTrigger (app.js:1:722865)
retrieveLookupItem @ plugin:obsidian-icon-folder:5411
eval @ plugin:obsidian-icon-folder:5430
computeNodesWithPath @ plugin:obsidian-icon-folder:5429
onMount @ plugin:obsidian-icon-folder:5435
eval @ plugin:obsidian-icon-folder:6603
handleChangeLayout @ plugin:obsidian-icon-folder:6601
eval @ plugin:obsidian-icon-folder:6458
e.tryTrigger @ app.js:1
e.trigger @ app.js:1
t.trigger @ app.js:1
(anonymous) @ app.js:1
l @ app.js:1
c @ app.js:1
setTimeout (async)
e.tryTrigger @ app.js:1
e.trigger @ app.js:1
t.trigger @ app.js:1
(anonymous) @ app.js:1
l @ app.js:1
c @ app.js:1
setTimeout (async)
u @ app.js:1
t.updateLayout @ app.js:1
r @ app.js:1
```

<img width="555" alt="Iconize fails to load" src="https://github.com/user-attachments/assets/55b7796e-d584-4f81-9c66-bcb774c41f18">

The 2nd error actually repeats itself when I interact with Obsidian by switching to another note from the files explorer, for example 😊 ...

I thought it could be due to some kind of conflict with another plugin I have in that vault but even if I disable all the other community plugins (leaving Iconize the only plugin enabled), I still get the same `Failed to load...` notification + errors in the console.

I tried to repro in a brand new empty vault and the sandbox vault but I couldn't for some unknown reasons 🤷‍♀️ ...

**Environment:** 
```
SYSTEM INFO:
	Obsidian version: v1.7.2
	Installer version: v1.6.7
	Operating system: Darwin Kernel Version 24.0.0: Mon Aug 12 20:52:12 PDT 2024; root:xnu-11215.1.10~2/RELEASE_ARM64_T6020 24.0.0
	Login status: logged in
	Language: en
	Catalyst license: supporter
	Insider build toggle: on
	Live preview: on
	Base theme: dark
	Community theme: none
	Snippets enabled: 4
	Restricted mode: off
	Plugins installed: 21
	Plugins enabled: 1
		1: Iconize v2.14.3

RECOMMENDATIONS:
	Custom theme and snippets: for cosmetic issues, please first try updating your theme and disabling your snippets. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community theme and snippets.
	Community plugins: for bugs, please first try updating all your plugins to latest. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community plugins.

```

Note that while there are errors in the console and the plugin "fails to load" ... It still seems to work 😅 .
(At least, I still get twemojis in place of the native MacOS emojis, as this is what I chose within the settings of Iconize 😊 )

Also, if I disable Iconize on Obsidian 1.7.2, I can't enable it again 🙃 : I need to downgrade to Obsidian 1.6.7, enable Iconize then upgrade again to Obsidian 1.7.2 🙃

**Comment by spiritualgeek** [comment]
@FlorianWoelki Experiencing the same as above

**Comment by holtwick** [comment]
+1 same here

**Comment by arcticpinecone** [comment]
+1 same for me:
![image](https://github.com/user-attachments/assets/7fb457f6-2d41-46ec-a5b4-159b0abeaa29)

**Comment by Drakemoor** [comment]
I have the same issue.

**Comment by matthewspear** [comment]
Also having the same issue on v1.7.2

**Comment by albertoloscerritos** [comment]
Same here

**Comment by YannMiro** [comment]
https://docs.obsidian.md/Plugins/Guides/Understanding+deferred+views

**Comment by Drakemoor** [comment]
> https://docs.obsidian.md/Plugins/Guides/Understanding+deferred+views

What does this mean!? Can we use it in someway to fix our issue?

**Comment by arcticpinecone** [comment]
> > https://docs.obsidian.md/Plugins/Guides/Understanding+deferred+views
> 
> 
> 
> What does this mean!? Can we use it in someway to fix our issue?
> 
> 

It's a tip to the maintainer of this plugin, that a breaking change could be due to this, and he linked to this page for review. So I imagine it will hopefully be fixed soon.

**Comment by YannMiro** [comment]
> > https://docs.obsidian.md/Plugins/Guides/Understanding+deferred+views
> 
> What does this mean!? Can we use it in someway to fix our issue?
>
> It's a tip to the maintainer of this plugin, that a breaking change could be due to this, and he linked to this page for review. So I imagine it will hopefully be fixed soon.

Yeah, I linked it in case the dev is unaware of the Obsidian updates or this project is not a priority right now

**Comment by gwecken222** [comment]
Same problem under Obsidian 1.7.3 and MacOS 15.

**Comment by phntmPwr** [comment]
Same with 17.3 on Windows.

**Comment by JSG-1** [comment]
I have the same issue.  i keep getting an error about — Failed to load plugin "obsidian-icon-folder"

**Comment by pedramamini** [comment]
interestingly, one of my vaults still works just fine. no idea how/why and what is different about the two setups.

**Comment by zui097** [comment]
The problem does not occur when the bookmark plugin is deactivated (maybe others are also affected).
There is an onload method which loops through other plugins and there something is going wrong.

**Comment by BrtCst** [comment]
> The problem does not occur when the bookmark plugin is deactivated (maybe others are also affected). There is an onload method which loops through other plugins and there something is going wrong.

Same here, disabling and reenabling Bookmarks core plugin does the trick.

**Comment by arcticpinecone** [comment]
Yes.. ok, I was skepitcal. Literally the moment I unticked Bookmark plugin, all the icons appeared instantly. 😲

**Comment by pedramamini** [comment]
@zui097 you're the man!

You can still re-enable bookmarks once the icons are restored.

**Comment by namtrah** [comment]
If you mean the Core Plugin *Bookmarks*. I tried this and it did not work for me. So still an issue?

**Comment by Cammagno** [comment]
> The problem does not occur when the bookmark plugin is deactivated (maybe others are also affected). There is an onload method which loops through other plugins and there something is going wrong.

It doesn't work for me.
Maybe I have some other plugin blocking this.
How did you find out about Bookmark one being the culprit for you?

**Comment by pedramamini** [comment]
I can confirm this works on my end too. Simply toggling off/on the core bookmarks plugin restores Iconize. I am also curious to hear the answer to Cammagno's question on "How did you find out about Bookmark one being the culprit for you?"

**Comment by Cammagno** [comment]
I've just find out that, without disabling and enabling again the Bookmark plugin, if you simply has the bookmark view active when loading obsidian (leaving it active when closing obsidian) let this plugin work.

Problem is, in this way, some other plugins don't (for example Folder Note)

BUT

good news is that it's enough to leave the default view enabled at startup to let all the other plugins start, then
switch to the Bookmarks view, switch off this plugin and then switch it on again.

**Comment by zui097** [comment]
> > The problem does not occur when the bookmark plugin is deactivated (maybe others are also affected). There is an onload method which loops through other plugins and there something is going wrong.
> 
> It doesn't work for me. Maybe I have some other plugin blocking this. How did you find out about Bookmark one being the culprit for you?

I had the same log like https://github.com/FlorianWoelki/obsidian-iconize/issues/566#issuecomment-2366744539
where BookmarkInternalPlugin is mentioned.

So I digged into main.ts of the iconize code (lines 381ff) and just tried to deactivate the BookbmarkPlugin to get iconize running again.

So maybe other plugins also are doing some thing in their onMount and register methods which should be catched here.

**Comment by I-Pch** [comment]
The `Bookmark` tab thing would explain why I couldn't repro the issue in a brand new empty vault (as I had no bookmarked note in there) or why it was happening in the vault where I encountered this, as I have 1 bookmarked note 😊 ...  

Quitting/re-launching Obsidian with the Bookmark tab open still produce errors within the console 😊... Different than the ones I was getting initially though... 

<img width="556" alt="Iconize - Obsidian 1 7 3" src="https://github.com/user-attachments/assets/1c317c70-4dcd-4147-9f71-c1f3cfc9b9dd">

Now, if I un-bookmark the bookmarked note in that vault quit & re-launch Obsidian (1.7.3) on the Files explorer tab, the `settings` error doesn't re-appear and I don't get the "Failed to load ..." notification and I can disable & enable Iconize as much as I want without having to rollback to the latest public release of Obsidian 😅 

<img width="556" alt="Iconize - Obsidian 1 7 3 - No bookmarked notes" src="https://github.com/user-attachments/assets/4a85d04d-e55a-441d-b628-e440ba491a88">

**Comment by FlorianWoelki** [comment]
Hey there,

Currently, I have a lot of stuff going on in my life, and this project was kind of de-prioritised. However, I am going to patch all the issues whenever the insider builds are fully released.

**Comment by albertoloscerritos** [comment]
Thanks mate 👍

**Comment by Cammagno** [comment]
> However, I am going to patch all the issues whenever the insider builds are fully released.

Thanks a lot!

BTW, Obsidian 7.4 has just been released

**Comment by clclementsj** [comment]
Awesome, thank you for your work on this very nice plugin. —CharlesOn Oct 15, 2024, at 4:07 AM, Florian Woelki ***@***.***> wrote:﻿
Hey there,
Currently, I have a lot of stuff going on in my life, and this project was kind of de-prioritised. However, I am going to patch all the issues whenever the insider builds are fully released.

**Comment by pyxelr** [comment]
> I've just find out that, without disabling and enabling again the Bookmark plugin, if you simply has the bookmark view active when loading obsidian (leaving it active when closing obsidian) let this plugin work.
> 
> Problem is, in this way, some other plugins don't (for example Folder Note)
> 
> BUT
> 
> good news is that it's enough to leave the default view enabled at startup to let all the other plugins start, then switch to the Bookmarks view, switch off this plugin and then switch it on again.

In my case (Obsidian 1.7.4), it is actually enough to open Obsidian (no icons), switch to Bookmarks view (the icons appear), switch back to Files view (the icons are there).

**Comment by FlorianWoelki** [comment]
_Status update:_
I've pushed some patches to fix the current issues with the deferred views so that the plugin even loads. There are still some issues, and fixes are needed.

**Comment by arcticpinecone** [comment]
Wanted to say thank you to you Florian, making the time for this, even with a lot going on. It is appreciated. I've been checking this thread frequently. 😁

**Comment by JSG-1** [comment]
Ditto!

Juan Salazar
CEO
US: 956.429.3400
MX: 800.681.8063
Assistant: 956.616.5420
www.mdcdatacenters.com

**Comment by FlorianWoelki** [comment]
I've got a fully working version now in the main branch. It would be nice if someone else could test it :)

**Comment by gfxholo** [comment]
Just tested the new build on Windows + Android. All basic features are working fine, including link icons :)

![Iconize for 1.7.4](https://github.com/user-attachments/assets/34dd3222-8f85-4fea-9f98-ae9d233f6773)

**Comment by Drakemoor** [comment]
> I've got a fully working version now in the main branch. It would be nice if someone else could test it :)

Thank you!

How do I update to this new version? It still shows the 2.14.3 for me and I don't see a beta? 

Thanks again for the quick and amazing job you are doing. 🙇‍♂️

**Comment by arcticpinecone** [comment]
> > I've got a fully working version now in the main branch. It would be nice if someone else could test it :)
> 
> Thank you!
> 
> How do I update to this new version? It still shows the 2.14.3 for me and I don't see a beta?
> 
> Thanks again for the quick and amazing job you are doing. 🙇‍♂️

This typescript project can be downloaded through git, and compiled locally on your own machine. However, I think the owner wanted more testers so that if it was suitable, they would then put that to a release which would then go to the Obsidian Plugin update. I was poking around with this earlier too, I should try compiling it. I've done it once for another plugin, but now I forgot how :D

**Comment by YannMiro** [comment]
> > I've got a fully working version now in the main branch. It would be nice if someone else could test it :)
> 
> Thank you!
> 
> How do I update to this new version? It still shows the 2.14.3 for me and I don't see a beta?
> 
> Thanks again for the quick and amazing job you are doing. 🙇‍♂️

I think that by installing the repo through BRAT you can achieve the same result. It is pretty straightforward.

**Comment by FlorianWoelki** [comment]
> > > I've got a fully working version now in the main branch. It would be nice if someone else could test it :)
> > 
> > 
> > Thank you!
> > How do I update to this new version? It still shows the 2.14.3 for me and I don't see a beta?
> > Thanks again for the quick and amazing job you are doing. 🙇‍♂️
> 
> I think that by installing the repo through BRAT you can achieve the same result. It is pretty straightforward.

Through BRAT it should be possible, yes.

**Comment by Drakemoor** [comment]
I've tried it through BRAT, but it doesn't show a version newer than 2.14.3. Thatnks for tring to help.

**Comment by arcticpinecone** [comment]
Yeah I too had tried BRAT and it didn't work, which is why I assumed the compiling. I must be missing something 🤷🏼

**Comment by arcticpinecone** [comment]
Alright, well you motivated me to do the full build route anyway. After compiling the main branch, replacing the files with the built ones, it was happily working alongside Bookmarks and worked without error. 

```
$ npm install
npm WARN deprecated @humanwhocodes/config-array@0.13.0: Use @eslint/config-array instead
npm WARN deprecated @humanwhocodes/object-schema@2.0.3: Use @eslint/object-schema instead
npm WARN deprecated eslint@8.57.1: This version is no longer supported. Please see https://eslint.org/version-support for other options.

> @florianwoelki/obsidian-iconize@0.0.1 prepare
> husky install

husky - Git hooks installed

added 663 packages, and audited 664 packages in 51s

152 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```

```
$ npm run build

> @florianwoelki/obsidian-iconize@0.0.1 build
> rollup --config rollup.config.js --environment BUILD:production


./src/main.ts → ....
copying ./main.js...
copying ./manifest.json...
copying ./src/styles.css...
created . in 23.8s
```

Thank you

**Comment by roman-balzer** [comment]
I'll second that. BRAT did not work for me, but could be a user error. Build from source works (With Bookmarks enabled)

**Comment by gfxholo** [comment]
BRAT can only install [GitHub releases](https://tfthacker.com/brat-developers#How+BRAT+works), so it won't pick up the new fixes in the repo yet.

> This typescript project can be downloaded through git, and compiled locally on your own machine. However, I think the owner wanted more testers so that if it was suitable, they would then put that to a release which would then go to the Obsidian Plugin update. I was poking around with this earlier too, I should try compiling it. I've done it once for another plugin, but now I forgot how :D

This is the correct answer for now. 💜

If you know how to use `npm` or `pnpm`, just follow the directions in [README > Development](https://github.com/FlorianWoelki/obsidian-iconize?tab=readme-ov-file#development) to get it running.

**Comment by DylanSpeiser** [comment]
+1, please fix soon

**Comment by FlorianWoelki** [comment]
This issue has been resolved in [2.14.4](https://github.com/FlorianWoelki/obsidian-iconize/releases/tag/2.14.4).

**Comment by Drakemoor** [comment]
Thank you so much for this. 🙇‍♂️

**Comment by BrtCst** [comment]
Thank you very much @FlorianWoelki !

**Comment by sicahjes** [comment]
> This issue has been resolved in [2.14.4](https://github.com/FlorianWoelki/obsidian-iconize/releases/tag/2.14.4).

I have just come back to Obsidian after a break, and I am experiencing this issue despite updating.

- I can't enable via toggling, I get the 'failed to load plugin' error. 
- When I click on the the app and click the enable button, the error appears again but it toggles on
- However, no settings cog comes up and the plugin does not appear to be active. 

**Obsidian**: 1.7.7
**Iconise**: 2.14.6

**Comment by BarthSarafin** [comment]
I also have this issue exactly as @sicahjes describes as well.

**Obsidian**: 1.8.10
**Iconise**: 2.14.7
